File: Problem Solving/Strange Counter.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'strangeCounter' function below.
#
# The function is expected to return a LONG_INTEGER.
# The function accepts LONG_INTEGER t as parameter.
#

def strangeCounter(t):
    total = getSum(t)
    return (total-t)

def getSum(t):
    cycle = getCycle(t)
    res = 4
    if(cycle == 1):
        return res;   
    i = 1
    while(i < cycle):
        res += 3*int((math.pow(2,i))) 
        i += 1
    return res

def getCycle(t):
    i = 0
    while(t > 0):
        t -= 3*int((math.pow(2,i)))
        i += 1
    return i  

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input().strip())

    result = strangeCounter(t)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Problem Solving/Strange Counter.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

class Result {

    /*
     * Complete the 'strangeCounter' function below.
     *
     * The function is expected to return a LONG_INTEGER.
     * The function accepts LONG_INTEGER t as parameter.
     */

    public static long strangeCounter(long t) {
        return(getSum(t) - t);

    }
    
    public static long getSum (long t){
        long cycle = getCycle(t);
        long sum = 4;
        if(cycle == 1){
            return sum;
        }
        for(int i = 1 ; i < cycle ; i++){
            sum += 3*getExp(i);
        }
        return sum;
        
    }
    
    public static long getCycle (long t){
        int i = 0;
        while(t > 0){
            t -= 3*getExp(i);
            i += 1;
        }
        return i;
    }
    public static long getExp (long e){
        long res = 1;
        for(int i = 0 ; i < e ; i++){
            res*=2;
        }
        return res;
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        long t = Long.parseLong(bufferedReader.readLine().trim());

        long result = Result.strangeCounter(t);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}

